ssetests/testapi_stream_server.go

```diff
@@ -0,0 +1,155 @@
+package ssetests
+
```

Comment:
Again the basic functionality here is mostly unchanged, it's just a cleaner API, but there is one other definite improvement.

Previously, there was always a maximum of one active connection per stream— there was a single reference to this in the server state, and if a second connection came in to the same endpoint before the first one ended, it would get confused about whether you were trying to send data to the first connection or the second one.

I reworked this so that you can call `AwaitConnection` any number of times and receive an independent `StreamConnection` each time. The latter is what you send data on. I deliberately did not implement any kind of multiplexing— as you would in a real SSE server where the same data is supposed to go out to every client at once— because I wanted the test to unambiguously control what the clients are getting. We don't currently have any tests that involve multiple concurrent connections to one stream, but it's a reasonable thing to do.